modules/launcher/src/main/java/org/mule/runtime/module/launcher/application/DefaultMuleApplication.java

```diff
@@ -172,24 +174,25 @@ public void init()
 
         try
         {
-            ConfigurationBuilder cfgBuilder = domainRepository.getDomain(descriptor.getDomain()).createApplicationConfigurationBuilder(this);
-            if (!cfgBuilder.isConfigured())
+            ArtifactMuleContextBuilder artifactBuilder = new ArtifactMuleContextBuilder()
+                    .setArtifactProperties(descriptor.getAppProperties())
+                    .setArtifactType(APP)
```

Comment:
May make sense what you say in the future. That's not the case right now. The only difference is the set of services that are available in each case plus a couple of restrictions that may be removed in the future. This is part of a big refactoring of the whole spring context creation and mule context as is only internal API. If there's a good reason to refactor the builder we can do it in the future.